portfolio/src/main/webapp/script.js

```diff
@@ -158,4 +162,16 @@ function login() {
 function start() {
 	checkLogin();
 	getComments();
+	blobUpload();
+}
+
+function blobUpload() {
+	fetch('/blobstore-upload-url')
+		.then((response) => {
+			return response.text();
+		})
+		.then((imageUploadUrl) => {
+			const commentForm = document.getElementById('form');
+			commentForm.action = imageUploadUrl;
```

Comment:
So, does this successfully change the form action on career.html to upload an image? After this happens, can the user still use the form to add a comment? I don't really understand how setting the action to an upload URL actually works.

Reply:
BlobStoreServlet is creating uploadUrl which has succespath "/data", uploadUrl becomes new action for the form. The way I understand, when submit button is clicked, the form first redirects to uploadUrl which is used by Blobstore API, it processes file and adds info about it to request object. After successfully doing that, it redirects to "/data" with the request that has info about blob. From here, as before, we can get each parameter of the form and get uploaded image URL from request object  (which is handled by method getUploadedFileUrl)

Reply:
Got it, thank you for explaining that!